### ISSUE TITLE
MaterialIcon component should not load the material icon font (blocks page load)

### ISSUE THREAD

**Issue by johnelliott** [material-components/material-components-web-react#98]
Hello, I'm using the code in an app and totally enjoying the library:

## about app
- Static site
- Hosted on firebase
- Bundled/transpiled with webpack/babel-loader
- Bundled using webpack rules to import SCSS directly from node_modules/@material/*

## issue: slow rendering (fonts)

I noticed this when I ran Lighthouse (with emoji commentary)
😑 Opportunities
￼
![image](https://user-images.githubusercontent.com/5633265/41195764-b9f892ae-6c01-11e8-83f2-09df514eccca.png)

￼
🤔 ok cool, go find in devtools…
￼
![image](https://user-images.githubusercontent.com/5633265/41195766-c1d8c7be-6c01-11e8-8181-a93184719c68.png)

🤔 ok, go to bundle…

found this CSS import in https://github.com/material-components/material-components-web-react/tree/master/packages/material-icon: `@import url(https://fonts.googleapis.com/icon?family=Material+Icons);`

🤔 ok, go get that resource in chrome…

```css
@font-face {
  font-family: 'Material Icons';
  font-style: normal;
  font-weight: 400;
  src: url(https://fonts.gstatic.com/s/materialicons/v38/flUhRq6tzZclQEJ-Vdg-IuiaDsNa.woff) format('woff');
}

.material-icons {
  font-family: 'Material Icons';
  font-weight: normal;
  font-style: normal;
  font-size: 24px;
  line-height: 1;
  letter-spacing: normal;
  text-transform: none;
  display: inline-block;
  white-space: nowrap;
  word-wrap: normal;
  direction: ltr;
}
```
💪 ok cool, `MaterialIcon` has sass with this fonts.gstatic.com URL.

Basically any of these fonts in css files that are in HEAD will block rendering on the network if the user puts a link to the css file in head or in a style tag.

So I pre-loaded https://fonts.googleapis.com/icon?family=Material+Icons in the head of my HTML, and could preload the woff file also, but that is not really a stable way to go about it since any of the packages in the chain could break without warning.

## try to fix issue
I was able to configure webpack to consume the @ material sass directly, but because of this URL I don't know how to make this code lighthouse-friendly without putting it after the body. I may be doing something wrong at a higher level.

## solution?
**TL;DR:** Could we come up with a way to bundle in the font resources? Or SVGs? Maybe that takes the form of documentation or an example.

My sense is that some users are going to write this off as being slow rather than tear in to the details, so if we could make stuff 'just work' as they say, it'd be great. Surely for places like the Top Navbar in-line SVGs would be more performant than an icon font. (I think it's ~62kb) Maybe I don't know about a bundler that would let me resolve this, the closest thing I found was 
https://www.npmjs.com/package/google-fonts-plugin, but I don't know how I'd connect it to this code as a user.

Where'd I go wrong here? Is there something on the roadmap that I'd want to know about?

**Comment by moog16** [comment]
@johnelliott thanks for filing the issue. I'll look into this and get back to you. I see the problem that you and others will have with the MaterialIcon component. Right now I'm leaning towards base64 encoding it into the Sass of MaterialIcon and using that google fonts plugin. I'll check in with the rest of the team to see what solutions they come up with. 

As a side note: I think that the 1st initial render is slow, but the subsequent renders are faster because Chrome/FF should be caching that font.

**Comment by moog16** [comment]
@johnelliott I briefly got some more ideas from the team, and I'm not sure if encoding the font is the correct way to go. 

1. You wouldn't get font updates or bug fixes until we synced with the new font file
2. It would bloat your css instead (I recognize that 1 large file download is better than many small downloads)

I think the google fonts plugin belongs in your build step and not ours. I understand this won't work for people using CRA, but I see CRA as a quick way to get going or spike an idea. 

Let me know if this helps.

**Comment by johnelliott** [comment]
Thanks Matt. I see your reply and I will answer once I get a moment.

**Comment by moog16** [comment]
@johnelliott spoke to @kfranqueiro. We both agree we should remove the url from the Sass file so that the end user (you) can have more control over how it is being imported. We came up with this solution:

1. Replace material-icon with [icon button](https://github.com/material-components/material-components-web/tree/master/packages/mdc-icon-button), which entail creating a react icon button component.
1. referencing the icon toggle readme to [preload](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content) the icon font.

We decided that icon toggle was the way to go because of this issue https://github.com/material-components/material-components-web/issues/2702 found in our core repo. Basically we need it to have a height and width. 

You're probably asking timeline: We are currently in the middle of a MDC Web version sync, which this can be included with.

**Comment by johnelliott** [comment]
@moog16 that sounds like a double-win. I just looked over the icon-button and material icons code for how it all works.

So if I'm understanding this correctly, it's
1. Removing [this import](https://github.com/material-components/material-components-web-react/blob/master/packages/material-icon/index.scss#L1) will eliminate a round-trip needed to get the font file _after_ the Google Web Fonts css is fetched. Then the user will have the option of implementing the [self hosting method here](https://google.github.io/material-design-icons/#icon-font-for-the-web) if desired.  The user becomes responsible for including the material-icons css either way.
2. solve material-components/material-components-web#2702

That sounds like a solid choice, it'll certainly solve my case if the import to Google Web Fonts service comes out. Thanks for looking in to it.

**Comment by moog16** [comment]
@johnelliott yes that's right! Working on that today. Hopefully we'll have the sync ready to go by the end of the week if all goes well.

**Comment by johnelliott** [comment]
That would be a good way around. This isn't a super-urgent matter because it's not yet deployed anywhere. :)

**Comment by lynnjepsen** [comment]
Steps to reproduce:
* Use MaterialIcon in a brand new react app
* Pull up the devtools and look at the Network

Expected Behavior
* There should be no network transaction for downloading https://fonts.gstatic.com/s/materialicons/*

Actual Behavior
* Network downloads https://fonts.gstatic.com/s/materialicons/*

**Comment by lynnjepsen** [comment]
When we update the MaterialIcon code so it does *not* download the material icon font...we should also provide documentation to our users on how to lazy load the material icon font.

**Comment by johnelliott** [comment]
Nice work :) I am looking forward to using the lib more.
